libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/roadlabel/RoadNameLabelComponent.kt

```diff
@@ -1,16 +1,29 @@
 package com.mapbox.navigation.dropin.component.roadlabel
 
 import android.view.View
+import com.mapbox.api.directions.v5.DirectionsCriteria
 import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.formatter.DistanceFormatterOptions
 import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.formatter.MapboxDistanceFormatter
 import com.mapbox.navigation.dropin.extensions.flowLocationMatcherResult
+import com.mapbox.navigation.dropin.extensions.flowRouteProgress
 import com.mapbox.navigation.dropin.lifecycle.UIComponent
+import com.mapbox.navigation.ui.maneuver.api.MapboxManeuverApi
+import com.mapbox.navigation.ui.maps.NavigationStyles
 import com.mapbox.navigation.ui.maps.roadname.view.MapboxRoadNameView
 import kotlinx.coroutines.flow.collect
 import kotlinx.coroutines.launch
 
 @OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
-class RoadNameComponent(private val roadNameView: MapboxRoadNameView) : UIComponent() {
+class RoadNameLabelComponent(
+    private val roadNameView: MapboxRoadNameView,
+    val maneuverApi: MapboxManeuverApi = MapboxManeuverApi(
```

Comment:
Let's use `MapboxRouteShieldApi` instead. This is the same API that `maneuver` uses internally

Reply:
Ok, I didn't know the internals.